GetIntoTeachingApi/Startup.cs

```diff
@@ -22,6 +24,10 @@ public Startup(IConfiguration configuration)
         // This method gets called by the runtime. Use this method to add services to the container.
         public void ConfigureServices(IServiceCollection services)
         {
+            services.AddScoped<INotificationClientAdapter, NotificationClientAdapter>();

```

Comment:
Worth a check here that AddScoped is really what you want.

Services, quite often are scoped as Singletons with AddSingleton to keep the same object alive across multiple requests.

It depends on the functionality within each individual service.

Reply:
Yeah that's a good shout, these could be singletons - I'll get that updated 👍